codegen.py

```diff
@@ -0,0 +1,127 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Bao Project and Contributors. All rights reserved
+
+import numpy as np
+import argparse
+import glob
+import os
+
+"""
+This script is used to generate Bao Project tests code.
+It searches for C source files with 'BAO_TEST' markers and creates corresponding test functions.
+"""
+
```

Comment:
@Diogo21Costa you should standardize the comments, either use **# Comment X** or **""" Comment X """**.  And please limit the column width in the whole file to 80 characters.

Reply:
Thank you for your review! I wanted to clarify the difference between the comment styles used in the code:

1. The triple-quoted (""" Comment X """) comment at the start is a module docstring. It provides an overview of the module's purpose.
2. The single-line comments (# Comment X) explain specific lines of code or steps for developers. In this case, they serve to specify the license.

Both styles serve distinct purposes, enhancing readability. Maintaining this differentiation helps convey both the module's intention and code specifics effectively.

In addition to addressing the comment, I've adjusted the code's line width to ensure it falls within the 80-character limit. Thank you for your consideration!

Reply:
Thank you @Diogo21Costa for the clarification. It makes sense now! No more concerns from my side. I'll approve.